languages/java/oso/src/test/java/com/osohq/oso/OsoTest.java

```diff
@@ -0,0 +1,110 @@
+package com.osohq.oso;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.net.URL;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+public class OsoTest {
+  protected Oso o;
+
+  public static class Actor {
+    public String name;
+
+    public Actor(String name) {
+      this.name = name;
+    }
+
+    public List<Company> companies() {
+      return List.of(new Company(1));
+    }
+  }
+
+  public static class Widget {
+    public int id;
+
+    public Widget(int id) {
+      this.id = id;
+    }
+  }
+
+  public static class Company {
+    public int id;
+
+    public Company(int id) {
+      this.id = id;
+    }
+
+    public String role(Actor a) {
+      if (a.name.equals("president")) {
+        return "admin";
+      }
+
+      return "guest";
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj instanceof OsoTest.Company && ((OsoTest.Company) obj).id == this.id;
```

Comment:
Does this pass if `equals` is only defined for `Company`? That would be how I think most people would want to write it.

Reply:
Do you mean if its only defined for `Company` and not the other two classes? Seems to, yeah. 😸